internal/provider/resource_host_set_plugin.go

```diff
@@ -131,6 +132,25 @@ func setFromHostSetPluginResponseMap(d *schema.ResourceData, raw map[string]inte
 		attrRaw, ok := raw["attributes"]
 		switch ok {
 		case true:
+			if attrMap, ok := attrRaw.(map[string]interface{}); ok {
+				for attrKey := range attrMap {
+					// Flatten attribute filters data structure from Boundary SDK to match terraform input data structure
```

Comment:
I thought about solving it generally by adjusting the data structure of the API/SDK to match what Boundary TF Provider expects and how the CLI even passes values but that could possibly break some existing integrations with our public API.

For restricting the change to just the AWS plugin, the host-catalog has the plugin name/type but I don't think the host-set returns that information. I can crosscheck

Reply:
Since the input from TF provider is user provided, I was thinking it could be tricky to change the data structure of that since users can enter almost anything. From the SDK, the data structure we get is consistent and deterministic so it's easier to change.

I can add a comment to explain why we are doing this and what plugin this change applies to so future developers have more context about this change.

I'll add some test coverage for it

Reply:
I can update it to be very targeted to just filters as you suggested earlier and also add a comment about why we need to flatten this for just aws

I tried to write some tests for the issue but it seems the only accepted `plugin_name` on Boundary TF is `loopback`. `aws` nor `azure` are not accepted. My guess is because of how the controller testing stuff is set up for Boundary. Only allows `loopback` for testing which does not have the issue we are trying to solve for here.
I manually tested it, however.